src/lexer.rs

```diff
@@ -365,6 +388,62 @@ impl<'a> Lexer<'a> {
 
         Some(self.text_token(start, TokenKind::Integer))
     }
+
+    fn read_decimal_part(&mut self, start: usize) -> Option<Token<'a>> {
+        self.read_char(); // consume the '.'
+
+        while self.peek_char().is_ascii_digit() {
+            self.read_char();
+        }
+
+        let p = self.peek_char();
+        match p {
+            'e' => self.read_exponent(start),
+            'E' => self.read_exponent(start),
```

Comment:
I can change this to match against `'e' | 'E'`, which will tidy things up a bit.

Reply:
SGTM